### PR TITLE
feat: Add USDT

### PR DESCRIPTION
**Pull request by martin-chainflip**
# Pull Request

Closes: PRO-1235

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [ ] I have updated documentation where appropriate.

### CODE REVIEW

**Comment by albert-llimos** [comment]
As I told Martin, there are some workarounds in the bouncer since the SDK doesn't support USDT yet. We wait for that and then update the bouncer accordingly before merging to main.

File: bouncer/commands/send_usdt.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env -S pnpm tsx
+// INSTRUCTIONS
+//
+// This command takes two arguments.
+// It will fund the ethereum address provided as the first argument with the amount
+// provided in the second argument. The asset amount is interpreted as USDT
+//
+// For example: ./commands/send_usdt.ts 0xcf1dc766fc2c62bef0b67a8de666c8e67acf35f6 1.2
+// will send 1.2 T to account 0xcf1dc766fc2c62bef0b67a8de666c8e67acf35f6
```
Comment (kylezs) [inline bouncer/commands/send_usdt.ts:9]:
```suggestion
// will send 1.2 USDT to account 0xcf1dc766fc2c62bef0b67a8de666c8e67acf35f6
```

File: bouncer/shared/new_swap.ts [hunk 2]
```diff
@@ -16,25 +24,54 @@ export async function newSwap(
   messageMetadata?: CcmDepositMetadata,
   brokerCommissionBps = defaultCommissionBps,
 ): Promise<void> {
+  //   const destinationAddress =
```
Comment (kylezs) [inline bouncer/shared/new_swap.ts:27]: ?

File: bouncer/shared/send.ts [hunk 3]
```diff
@@ -42,6 +42,17 @@ export async function send(asset: Asset, address: string, amount?: string, log =
       );
       break;
     }
+    case 'USDT': {
```
Comment (kylezs) [inline bouncer/shared/send.ts:45]: no need to dupe this right? can just do:
```
case 'USDC':
case 'USDT':
...
```

Comment (kylezs) [reply bouncer/shared/send.ts:45]: same for FLIP

File: bouncer/shared/swapping.ts [hunk 4]
```diff
@@ -175,9 +175,9 @@ export async function testAllSwaps() {
     messageMetadata?: CcmDepositMetadata,
   ) {
     if (destAsset === 'BTC') {
-      Object.values(btcAddressTypes).forEach((btcAddrType) => {
-        allSwaps.push(functionCall(sourceAsset, destAsset, btcAddrType, messageMetadata));
-      });
+      // Object.values(btcAddressTypes).forEach((btcAddrType) => {
```
Comment (kylezs) [inline bouncer/shared/swapping.ts:178]: ?

File: bouncer/shared/swapping.ts [hunk 5]
```diff
@@ -189,24 +189,26 @@ export async function testAllSwaps() {
...
-    ).toString(),
-  );
-  await approveTokenVault(
-    'FLIP',
-    (
-      BigInt(amountToFineAmount(defaultAssetAmounts('FLIP'), assetDecimals('FLIP'))) * 9n
-    ).toString(),
-  );
+  // await approveTokenVault(
+  //   'USDC',
```
Comment (kylezs) [inline bouncer/shared/swapping.ts:193]: ?

File: state-chain/custom-rpc/src/lib.rs [hunk 6]
```diff
@@ -1398,7 +1398,9 @@ mod test {
 					(Asset::Btc, 0),
 					(Asset::Flip, 1000000000000000000),
 					(Asset::Usdc, 0),
+					(Asset::Usdt, 0),
 					(Asset::Dot, 0),
+					(Asset::Usdt, 0),
```
Comment (kylezs) [inline state-chain/custom-rpc/src/lib.rs:1403]: why is this twice?

File: state-chain/custom-rpc/src/lib.rs [hunk 7]
```diff
@@ -1425,13 +1427,16 @@ mod test {
 					(Asset::Btc, 0),
 					(Asset::Flip, u128::MAX / 2),
 					(Asset::Usdc, 0),
+					(Asset::Usdt, 0),
 					(Asset::Dot, 0),
+					(Asset::Usdt, 0),
```
Comment (kylezs) [inline state-chain/custom-rpc/src/lib.rs:1432]: same^

File: state-chain/custom-rpc/src/lib.rs (hunk 6 above)
Comment (martin-chainflip) [reply state-chain/custom-rpc/src/lib.rs:1403]: merge gone bad

File: bouncer/shared/new_swap.ts (hunk 2 above)
Comment (albert-llimos) [reply bouncer/shared/new_swap.ts:27]: there are a bunch of commented out things as workaround for the SDK. They shouldn't be there when I push the newest changes.

**Pull request merged**